### PR TITLE
Add ligo-segments

### PR DESCRIPTION
**Pull request by duncanmmacleod**
[`ligo-segments`](https://pypi.org/project/ligo-segments/) provides representations of semi-open intervals, typically used for representing time intervals.

### CODE REVIEW

**Comment by duncanmmacleod** [comment]
the appveyor build is failing with the following message:

```
ValueError: variant config in C:\projects\staged-recipes\recipes\ligo-segments\conda_build_config.yaml is ambiguous because it does not fully implement all zipped keys, or specifies a subspace that is not fully implemented.
```

I have no idea what this means, so some help from a knowledgable person would be appreciated. I can only guess that one or more of the python versions specified in the `conda_build_config.yaml` file aren't available on windows?

**Comment by CJ-Wright** [comment]
Is this PR relevant to this issue? https://github.com/conda-forge/conda-smithy/pull/786
@conda-forge/core

**Comment by isuruf** [comment]
We don't support python 3.4 anymore. You can remove the `conda_build_config.yaml` file and it should build for 2.7,3.5,3.6

If you are curious, the error is because we have,

```
c_compiler:
   - vs2008
   - vs2015
   - vs2015
python
   - 2.7
   - 3.5
   - 3.6
zip_keys:
-  - c_compiler
   - python
```
in our global config file and you overriding python here makes the 2 lists `python` and `c_compiler` have 2 different sizes.

**Comment by duncanmmacleod** [comment]
@isuruf, is this PR mergeable now?

File: recipes/ligo-segments/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  sha256: eab881e7646de8fb47ad7f602a8149488e938af73d5f639e65b7dfa2b7180657
+  patches:
+    - pure-python-windows.patch  # [win]
+
+build:
+  number: 0
+  script: python -m pip install --no-deps --ignore-installed .
+
+requirements:
+  build:
```
Comment (scopatz) [inline recipes/ligo-segments/meta.yaml]: I think you need to add a `{{ compiler('c') }}` dependency: https://conda-forge.org/docs/meta.html?#compilers

**Comment by duncanmmacleod** [comment]
@scopatz, is this ready now? Apologies if its not the done thing to nag about PRs (I know there are a lot of PRs for this project).

**Comment by scopatz** [comment]
This looks good to me! Unfortunately, merging PRs is blocked right now due to feedstock creation failing. We'll get to this ASAP

**Comment by duncanmmacleod** [comment]
Ah, ok, no problems, I just wanted to check if there were any more changes that needed to be made. Thanks.

**Comment by duncanmmacleod** [comment]
Thanks @scopatz.

**Pull request merged**